tools/autograd/gen_variable_type.py

```diff
@@ -218,15 +228,15 @@
 SAVE_OPTIONALTENSORLIST_IMPL = CodeTemplate("""\
 std::vector<c10::intrusive_ptr<TensorImpl>> ${tensorlist_name}_impl_saved(${tensorlist_name}.size());
 for (size_t i=0; i<${tensorlist_name}.size(); i++) {
-  c10::optional<Tensor> t = ${tensorlist_name}[i];
-  if (t.has_value() && t->defined()) ${tensorlist_name}_impl_saved[i] = t->getIntrusivePtr();
+  auto& t = ${tensorlist_name}[i];
+  if (t.has_value()) ${tensorlist_name}_impl_saved[i] = t->getIntrusivePtr();
```

Comment:
Is it always ok to remove the defined check here?

Reply:
Ah, nice catch! It's not ok (it was when only `OptionalTensorRef` got to this point).